presto-raptorx/src/main/java/com/facebook/presto/raptorx/chunkstore/ChunkStoreCleaner.java

```diff
@@ -0,0 +1,229 @@
+ */
+package com.facebook.presto.raptorx.chunkstore;
+
+import com.facebook.presto.raptorx.util.Database;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.concurrent.BoundedExecutor;
+import io.airlift.log.Logger;
+import io.airlift.stats.CounterStat;
+import io.airlift.units.Duration;
+import org.jdbi.v3.sqlobject.customizer.Bind;
+import org.jdbi.v3.sqlobject.customizer.BindList;
+import org.jdbi.v3.sqlobject.customizer.Define;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+import org.jdbi.v3.sqlobject.statement.SqlUpdate;
+import org.weakref.jmx.Managed;
+import org.weakref.jmx.Nested;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Set;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.facebook.presto.raptorx.util.DatabaseUtil.createJdbi;
+import static com.google.common.collect.Sets.difference;
+import static com.google.common.collect.Sets.newConcurrentHashSet;
+import static io.airlift.concurrent.Threads.daemonThreadsNamed;
+import static java.lang.Math.min;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.CompletableFuture.runAsync;
+import static java.util.concurrent.Executors.newCachedThreadPool;
+import static java.util.concurrent.Executors.newSingleThreadScheduledExecutor;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+public class ChunkStoreCleaner
+{
+    private static final Logger log = Logger.get(ChunkStoreCleaner.class);
+
+    private static final int CHUNK_BATCH_SIZE = 1000;
+
+    private final ScheduledExecutorService scheduler = newSingleThreadScheduledExecutor(daemonThreadsNamed("chunk-store-cleaner"));
+    private final ExecutorService coreExecutor;
+    private final Executor executor;
+    private final ChunkStore chunkStore;
+    private final Duration interval;
+    private final CleanerDao dao;
+    private final AtomicBoolean started = new AtomicBoolean();
+    private final CounterStat jobErrors = new CounterStat();
+    private final CounterStat deleteFailures = new CounterStat();
+    private final CounterStat deleteSuccesses = new CounterStat();
+
+    @Inject
+    public ChunkStoreCleaner(ChunkStore chunkStore, Database database, ChunkStoreCleanerConfig config)
+    {
+        this(chunkStore, database, config.getInterval(), config.getThreads());
+    }
+
+    public ChunkStoreCleaner(ChunkStore chunkStore, Database database, Duration interval, int threads)
+    {
+        this.chunkStore = requireNonNull(chunkStore, "chunkStore is null");
```

Comment:
"highker on Jan 30  Member
maybe moving the initialization to where it is declared"

This is the same across all classes. And the constructor is used in uni-test. If need change, we'd better do it as a whole refactor task another time.